sklearn/gaussian_process/kernels.py

```diff
@@ -33,6 +33,21 @@
 from sklearn.externals.funcsigs import signature
 
 
+def _check_length_scale(X, length_scale):
+    if np.iterable(length_scale):
```

Comment:
And I don't really get why we're accepting a 1d array of length 1 as a substitute for a scalar anyway...